test/framework/src/main/java/org/elasticsearch/test/discovery/MockZenPing.java

```diff
@@ -40,7 +41,9 @@
  * A {@link ZenPing} implementation which returns results based on an static in-memory map. This allows pinging
  * to be immediate and can be used to speed up tests.
  */
-public final class MockZenPing extends AbstractLifecycleComponent implements ZenPing {
+public final class MockZenPing extends AbstractComponent implements ZenPing {
+
+    public static class TestPlugin extends Plugin {}
```

Comment:
if we keep this, can we add a comment about what it does? I think it will confuse people like this.